resources/lib/lcdbase.py

```diff
@@ -497,15 +501,37 @@ def Render(self, mode, bForce):
     while (outLine < int(self.GetRows()) and inLine < len(self.m_lcdMode[mode])):
       #parse the progressbar infolabel by ourselfs!
       if self.m_lcdMode[mode][inLine]['type'] == LCD_LINETYPE.LCD_LINETYPE_PROGRESS:
-        # get playtime and duration and convert into seconds
-        percent = InfoLabel_GetProgressPercent()
-        pixelsWidth = self.SetProgressBar(percent, self.m_lcdMode[mode][inLine]['endx'])
-        line = "p" + str(pixelsWidth)
+        if mode == LCD_MODE.LCD_MODE_VOLCHANGING:
+          # get volume percentage
+          percent = InfoLabel_GetVolumePercent()
+          if InfoLabel_IsMuted():
+            percent = 0
+          endx = self.m_lcdMode[mode][inLine]['endx']
+          pixelsWidth = self.SetProgressBar(percent, endx)
+          line = str(pixelsWidth)
+          log(xbmc.LOGDEBUG, "Volume changing: %s: %d%%, Muted: %s" % (InfoLabel_GetInfoLabel("Player.Volume"), percent, InfoLabel_GetInfoLabel("Player.Muted")))
+          log(xbmc.LOGDEBUG, "Volume changing: line: %s, endx: %d" % (line, endx))
+        else:
+          # get playtime and duration and convert into seconds
+          percent = InfoLabel_GetProgressPercent()
+          pixelsWidth = self.SetProgressBar(percent, self.m_lcdMode[mode][inLine]['endx'])
+          line = "p" + str(pixelsWidth)
       else:
         if self.m_lcdMode[mode][inLine]['type'] == LCD_LINETYPE.LCD_LINETYPE_ICONTEXT:
           self.SetPlayingStateIcon()
 
-        srcline = InfoLabel_GetInfoLabel(self.m_lcdMode[mode][inLine]['text'])
+        srcline = self.m_lcdMode[mode][inLine]['text']
+        # Handle special player volume level
+        if self.m_vPythonVersion < (2, 7):
```

Comment:
I'm going to drop all compat to Python<2.7, Leia won't compile on any such old distro which brings such outdated python version at all...